pkg/knownhosts/knownhosts.go

```diff
@@ -0,0 +1,350 @@
+package knownhosts
+
+import (
+	"bufio"
+	"bytes"
+	"crypto/hmac"
+	"crypto/sha1"
+	"encoding/base64"
+	"errors"
+	"fmt"
+	"io"
+	"net"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/flynn/flynn/Godeps/_workspace/src/golang.org/x/crypto/ssh"
+)
+
+type KnownHosts struct {
+	Lines []*Line
+}
+
+func Unmarshal(in io.Reader) (*KnownHosts, error) {
+	k := &KnownHosts{
+		Lines: []*Line{},
+	}
+	s := bufio.NewScanner(in)
+	for s.Scan() {
+		l, err := parseHosts(s.Text())
+		if err == nil && l != nil {
+			k.Lines = append(k.Lines, l)
+		} else {
+			fmt.Println(err)
```

Comment:
Don't print in non-`main` packages, probably best to return a list of lines that failed as a composite error?